jwala-services/src/main/java/com/cerner/jwala/service/media/impl/MediaServiceImpl.java

```diff
@@ -85,8 +90,20 @@ public JpaMedia create(final Map<String, String> mediaDataMap, final Map<String,
     @Transactional
     public void remove(final String name) {
         final JpaMedia jpaMedia = mediaDao.find(name);
-        repositoryService.delete(jpaMedia.getLocalPath().getFileName().toString());
+        checkForExistingAssociationsBeforeRemove(name);
         mediaDao.remove(jpaMedia);
+        repositoryService.delete(jpaMedia.getLocalPath().getFileName().toString());
+    }
+
+    private void checkForExistingAssociationsBeforeRemove(String name) {
+        List<Jvm> jvmList = jvmPersistenceService.getJvms();
+        for (Jvm jvm : jvmList) {
+            if (jvm.getJdkMedia().getName().equalsIgnoreCase(name)) {
```

Comment:
Since we allow null in jdkmedia, it might throw NPE. change it to 
name.equalsIgnoreCase(jvm.getJdkMedia().getName())